loader.js

```diff
@@ -23,16 +23,31 @@ module.exports = async function mdxEnhancedLoader(src) {
   }
   // Scan for plugin `scan` option to return results based on RegEx patterns provided in config
   const scans = scanContent(options, content)
-  // Get file path relative to project root
-  const resourcePath = normalizeToUnixPath(this.resourcePath)
-    .replace(
-      normalizeToUnixPath(
-        path.join(normalizeToUnixPath(this.rootContext), 'pages')
-      ),
-      ''
-    )
-    .substring(1)
+  
+  // The next steps serve to support placing pages under `src/pages`:
+  // 1. Checks whether `src/pages` exists in current resource path
+  const usesSrc = path.normalize(this.resourcePath).includes(path.normalize("/src/pages/"));
+  
+  const normalizedResourcePath = normalizeToUnixPath(this.resourcePath);
+
+  let pageDirectoryPath = '';
+
+  if(usesSrc) {
+    // 2. Add `src/pages` to root path
+    pageDirectoryPath = normalizeToUnixPath(
+      path.join(normalizeToUnixPath(this.rootContext), 'src/pages')
+    );
+  } else {
+    pageDirectoryPath = normalizeToUnixPath(
+      path.join(normalizeToUnixPath(this.rootContext), 'pages')
+    );
+  }
 
+  // Get file path relative to project root
+  const resourcePath = normalizedResourcePath.replace(
+     pageDirectoryPath, ''
+    ).substring(1)
+  
```

Comment:
Same here, you can use the loader `options` to pass logic from `index.js` so that it doesn't need to be duplicated here

Reply:
I've modified these to take `usesSrc` from the options/pluginOptions but how do I do determine `usesSrc`  even before the `webpack()` config? I've got:
```js
// 1. Check whether `src/pages` exists
  const usesSrc = fs.existsSync(path.join(WHAT TO PASS HERE???, '/src/pages'));
  pluginOptions.usesSrc = usesSrc;
```